lib/ember_cli/shell.rb

```diff
@@ -37,6 +37,7 @@ def install
         exec "#{paths.bundler} install"
       end
 
+      exec "#{paths.ember} version || rm -rf #{paths.npm_deps} #{paths.bower_deps}"
```

Comment:
While this style of invocation is compact and UNIX-y, what do you think about implementing it in Ruby, in a platform-independent manner?

I'm thinking something like:

```rb
if invalid_ember_dependencies?
  clean_dependencies!
end

# ...
private

def invalid_ember_dependencies?
  # this might return the exit status -- I'm not sure.
  # if it does, this method would check for non-zero
  ! exec("#{paths.ember} version")
end

def clean_dependencies!
  [
    paths.node_modules, 
    paths.bower_components,
  ].map(&:rmtree)
end
```